File: code/kauthenticity/week4/prog_92344.js
function solution(board, skill) {
    var answer = 0;
    const [n, m] = [board.length, board[0].length];
    const tempBoard = Array(n + 1)
        .fill()
        .map(() => Array(m + 1).fill(0));

    skill.forEach(([type, r1, c1, r2, c2, degree]) => {
        tempBoard[r1][c1] += type === 2 ? degree : -degree;
        tempBoard[r1][c2 + 1] += type === 2 ? -degree : degree;

        tempBoard[r2 + 1][c1] += type === 2 ? -degree : degree;
        tempBoard[r2 + 1][c2 + 1] += type === 2 ? degree : -degree;
    });

    for (let i = 0; i < n; i++) {
        for (let j = 0; j < m - 1; j++) {
            tempBoard[i][j + 1] += tempBoard[i][j];
        }
    }

    for (let j = 0; j < m; j++) {
        for (let i = 0; i < n - 1; i++) {
            tempBoard[i + 1][j] += tempBoard[i][j];
        }
    }

    for (let i = 0; i < n; i++) {
        for (let j = 0; j < m; j++) {
            board[i][j] += tempBoard[i][j];

            if (board[i][j] > 0) {
                answer++;
            }
        }
    }
    return answer;
}

File: code/jonghyeonjo99/week4/prog_92344.py
def solution(board, skill):
    answer = 0
    imos = [[0 for _ in range(len(board[1]) + 1)] for _ in range(len(board) + 1)]

    for type, r1, c1, r2, c2, degree in skill:
        if type == 1:
            imos[r1][c1] -= degree
            imos[r2 + 1][c1] += degree
            imos[r1][c2 + 1] += degree
            imos[r2 + 1][c2 + 1] -= degree
        else: #type == 2
            imos[r1][c1] += degree
            imos[r2 + 1][c1] -= degree
            imos[r1][c2 + 1] -= degree
            imos[r2 + 1][c2 + 1] += degree
    
    #누적합 두번
    for i in range(len(board) + 1):
        for j in range(1,len(board[1]) + 1):
            imos[i][j] += imos[i][j-1]
    
    for j in range(len(board[1]) + 1):
        for i in range(1,len(board) + 1):
            imos[i][j] += imos[i-1][j]
    
    #board에 적용
    for i in range(len(board)):
        for j in range(len(board[1])):
            board[i][j] += imos[i][j]
            if board[i][j] > 0:
                answer += 1
            
    return answer